src/utils.jl

```diff
@@ -50,116 +50,99 @@ accuracy(nlp::T) where {T <: AbstractKnetNLPModel} =
+  v::AbstractVector{T},
   index::Int,
-) where {T <: Number}
-  sizecuArray = reduce(*, size(cuArray))
-  copyto!(cuArray, v[(index + 1):(index + sizecuArray)])
-  return sizecuArray
+) where {T <: Number, N}
+  sizearray = reduce(*, size(array))
+  copyto!(array, v[(index + 1):(index + sizearray)])
+  return sizearray
 end
 
 """
-    build_nested_array_from_vec(chain_ANN :: C, v :: Vector{T}) where {C <: Chain, T <: Number}
-    build_nested_array_from_vec(model :: KnetNLPModel{T, S, C}, v :: Vector{T}) where {T, S, C}
-    build_nested_array_from_vec(nested_array :: Vector{CuArray{T, N, CUDA.Mem.DeviceBuffer} where N}, v :: Vector{T}) where {T <: Number}
+    build_nested_array_from_vec(chain_ANN :: C, v :: AbstractVector{T}) where {C <: Chain, T <: Number}
+    build_nested_array_from_vec(model :: KnetNLPModel{T, S, C, V}, v :: AbstractVector{T}) where {T, S, C, V}
+    build_nested_array_from_vec(nested_array::AbstractVector{<:AbstractArray{T,N} where {N}}, v::AbstractVector{T}) where {T <: Number}
 
-Build a vector of `CuArray` from `v` similar to `Knet.params(model.chain)`, `Knet.params(chain_ANN)` or `nested_array`.
-Call iteratively `build_layer_from_vec` to build each intermediate `CuArray`.
+Build a vector of `AbstractArray` from `v` similar to `Knet.params(model.chain)`, `Knet.params(chain_ANN)` or `nested_array`.
+Call iteratively `build_layer_from_vec` to build each intermediate `AbstractArray`.
 This method is not optimized; it allocates memory.
 """
-build_nested_array_from_vec(model::T, v::Vector{T}) where {T <: AbstractKnetNLPModel} =
+build_nested_array_from_vec(model::K, v::AbstractVector{T}) where {T <: Number, S, K <: AbstractKnetNLPModel{T, S}} =
   build_nested_array_from_vec(model.chain, v)
 
-function build_nested_array_from_vec(chain_ANN::C, v::Vector{T}) where {C <: Chain, T <: Number}
+function build_nested_array_from_vec(chain_ANN::C, v::AbstractVector{T}) where {C <: Chain, T <: Number}
   param_chain = params(chain_ANN) # :: Param
   size_param = mapreduce((var_layer -> reduce(*, size(var_layer))), +, param_chain)
   size_param == length(v) || error(
     "Dimension of Vector v mismatch, function build_nested_array $(size_param) != $(length(v))",
   )
 
   param_value = (x -> x.value).(param_chain) # :: Vector{CuArray}
-  vec_CuArray = build_nested_array_from_vec(param_value, v)
-  return vec_CuArray
+  vec_cuarray = build_nested_array_from_vec(param_value, v)
+  return vec_cuarray
 end
 
 function build_nested_array_from_vec(
-  nested_array::Vector{CuArray{T, N, CUDA.Mem.DeviceBuffer} where N},
-  v::Vector{T},
+  nested_array::AbstractVector{<:AbstractArray{T,N} where {N}},
+  v::AbstractVector{T},
 ) where {T <: Number}
-  vec_CuArray = map(i -> similar(nested_array[i]), 1:length(nested_array))
-  build_nested_array_from_vec!(vec_CuArray, v)
-  return vec_CuArray
+  vec_cuarray = map(i -> similar(nested_array[i]), 1:length(nested_array))
+  build_nested_array_from_vec!(vec_cuarray, v)
+  return vec_cuarray
```

Comment:
```suggestion
  return array
```

Reply:
rested?

Reply:
Sorry, I meant `nested_array`.

Reply:
That's already the name of one of the arguments.

Reply:
complicated ... but ok

Reply:
I edited the previous message, I corrected the line
```julia
  similar_nested_array = map(array -> similar(array), nested_array)
# similar_nested_array = map(array -> similar(nested_array[array]), nested_array) #previously
```